apps/etl-func/src/domain/message-status.ts

```diff
@@ -23,27 +24,41 @@ const statusEnum = z.enum([
   "REJECTED",
 ]);
 
-const rejectedMessageStatusSchema = z.object({
-  fiscalCode: fiscalCodeSchema,
-  isArchived: z.boolean().default(false),
-  isRead: z.boolean().default(false),
-  messageId: z.string().ulid(),
-  rejection_reason: rejectionReasonSchema,
-  status: statusEnum.extract(["REJECTED"]),
-  updatedAt: z.number(),
-});
-
-const notRejectedMessageStatusSchema = z.object({
-  fiscalCode: fiscalCodeSchema,
-  isArchived: z.boolean().default(false),
-  isRead: z.boolean().default(false),
-  messageId: z.string().ulid(),
-  status: statusEnum.exclude(["REJECTED"]),
-  updatedAt: z.number(),
-});
+const messageStatusIdSchema = z.string().min(1);
 
-export const messageStatusSchema = z.discriminatedUnion("status", [
-  notRejectedMessageStatusSchema,
-  rejectedMessageStatusSchema,
-]);
+export const messageStatusSchema = z
+  .object({
+    fiscalCode: fiscalCodeSchema,
+    id: messageStatusIdSchema,
+    isArchived: z.boolean().default(false),
+    isRead: z.boolean().default(false),
+    messageId: z.string().ulid(),
+    updatedAt: z.number(),
+    version: z.number().gte(0),
+  })
+  .and(
+    z.discriminatedUnion("status", [
+      z.object({
+        status: statusEnum.exclude(["REJECTED"]),
+      }),
+      z.object({
+        rejection_reason: rejectionReasonSchema,
+        status: statusEnum.extract(["REJECTED"]),
+      }),
+    ]),
+  );
 export type MessageStatus = z.TypeOf<typeof messageStatusSchema>;
+
+export const messageStatusEventSchema = z.object({
```

Comment:
I would move the `messageEventStatusSchema` outside of message-status.ts and to it's own file like we did for the `messageEventSchema`

Reply:
[05f22d2](https://github.com/pagopa/io-messages/pull/62/commits/05f22d28c1406230f77b091c587067c0482a4662)